### DISCUSSION TITLE
Game never starts

### DISCUSSION THREAD

**Discussion by loudonadam** [AdriaGual/marvel-snap-bot#42] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/88801032/216139077-36a37552-cd97-491c-8472-e064f97b0966.png)
![image](https://user-images.githubusercontent.com/88801032/216139203-5d50b694-0fcc-47d5-98c5-3d84ba0539f0.png)

I'm pretty sure I've got all the connections set up. I have the new images of the play button and such from the update. The tmp file even refreshes with an image from the game when I run it, but it just never gets going. I'm not sure what step I'm missing. Any help from people who have been able to get this thing running would be super helpful, thank!

**Comment by AdriaGual** [comment, marked as answer]
Ok, there's an Announcement on the discussions talking about that.
https://github.com/AdriaGual/marvel-snap-bot/discussions/34

But i'll explain a bit more:
Yesterday, Marvel Snap made a patch update (the 11.18.1), in this patch, they changed the workflow of the matches (they've removed the 'collect rewards' button) AND they have changed the appearance of the PLAY BUTTON and NEXT BUTTON, is slight, but it's different.

SO you may be having one of this 2 problems:
- Possible problem 1:
You have outdated code (you haven't pull the commits from yesterday) and the bot is trying to find the old images on the new version of Marvel Snap.
- Possible problem 2:
You have the latest code but you haven't updated Marvel Snap to the 11.18.1 patch so the bot is trying to find the new images on the old app version.

Hope it helps!

**Reply by loudonadam** [reply]
Yeah, I reviewed that comment. Unfortunately, the game is up to date and I have a version of the code from just this morning.

**Reply by AdriaGual** [reply]
Hmm, you can execute the `python print_hand_cards.py`, this script will open a screenshot in real time of the bottom part of your emulator, then just grab the snipping tool you have (I use the default one windows10 has) and take a screenshot of the play button.

After that, go to the images folder and replace the play_button and go to the turns folder inside and replace it again.

Then play a match and do the same but for the next button (when a match has ended)

**Reply by loudonadam** [reply]
That worked and it will now start the match, which is awesome. And it's getting in a lot of the information, but it's bugging out on the first turn still. I tried looking at the function it's referencing, but couldn't figure out what is going wrong.

`[get_turn] Time elapsed: 0:00:00.027044
[get_turn] Time elapsed: 0:00:00.025694
[get_turn] Time elapsed: 0:00:00.025236
---------------------------
Picture: 0
* Turn: 1
[get_mana] Time elapsed: 0:00:00.004526
* Mana: 1
[get_my_hand_cards] Time elapsed: 0:00:02.133107
Hand_cards:
antman
blade
enchantress
odin
[get_fields] Time elapsed: 0:00:00.617822
* Active fields:
['first_field: wakandan_embassy', 'second_field: next_turn', 'third_field: next_2_turns']
[get_player_fields_cards] Time elapsed: 0:00:00.001869
* Player field cards:
---------------------------
| |-|-| || |-|-| || |-|-| |
| |-|-| || |-|-| || |-|-| |
---------------------------
Traceback (most recent call last):
  File "start.py", line 65, in <module>
    play_info = info.get_info(
  File "C:\Users\adam\marvel-snap-bot\info.py", line 41, in get_info
    cv2.waitKey(10)
cv2.error: OpenCV(4.6.0) D:\a\opencv-python\opencv-python\opencv\modules\highgui\src\window.cpp:1333: error: (-2:Unspecified error) The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Cocoa support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function 'cvWaitKey'`

**Reply by AdriaGual** [reply]
This error happens with opencv 4.6.0.

Install this:
`pip install opencv-python==4.7.0.68`
`pip install opencv-python-headless==4.6.0.66`

**Reply by loudonadam** [reply]
That worked and now its running through a couple games already. Thanks!

**Reply by AdriaGual** [reply]
Great to hear that!
